packages/permissions/src/hooks/use-is-authorized/use-is-authorized.spec.tsx

```diff
@@ -187,4 +208,271 @@ describe('when only one permission matches', () => {
+            ],
+            allAppliedActionRights: [
+              {
+                group: 'products',
+                name: 'canEditPrices',
+                value: true,
+              },
+            ],
+            demandedPermissions: ['ViewProducts'],
+            demandedActionRights: [{ group: 'products', name: 'EditPrices' }],
+            shouldMatchSomePermissions: true,
+          });
+
+          expect(queryByText('Is authorized: Yes')).toBeInTheDocument();
+        });
+      });
+      describe('if can edit prices on orders group', () => {
+        it('should pass isAuthorized as "false"', () => {
+          const { queryByText } = testRender({
+            allAppliedPermissions: [
+              { name: 'canManageProjectSettings', value: true },
+              { name: 'canViewProducts', value: true },
+            ],
+            allAppliedActionRights: [
+              {
+                group: 'orders',
+                name: 'canEditPrices',
+                value: true,
+              },
+            ],
+            demandedPermissions: ['ViewProducts'],
+            demandedActionRights: [{ group: 'products', name: 'EditPrices' }],
+            shouldMatchSomePermissions: true,
+          });
+
+          expect(queryByText('Is authorized: No')).toBeInTheDocument();
+        });
+      });
+    });
+  });
+
+  describe('dataFences', () => {
+    describe('with general permission', () => {
+      describe('when general permissions is manage', () => {
+        it('should indicate as authorized', () => {
+          const { queryByText } = testRender({
+            shouldMatchSomePermissions: true,
+            allAppliedPermissions: [{ name: 'canManageOrders', value: true }],
+            allAppliedDataFences: [
+              {
+                __typename: 'StoreDataFence',
+                type: 'store',
+                group: 'orders',
+                name: 'canViewOrders',
+                value: 'store-1',
+              },
+            ],
+            demandedPermissions: ['ManageOrders'],
+            demandedDataFences: [
+              {
+                type: 'store',
+                group: 'orders',
+                name: 'ManageOrders',
+              },
+            ],
+            selectDataFenceData: ({ type }) => {
+              switch (type) {
+                case 'store':
+                  return ['store-1'];
+                default:
+                  return null;
+              }
+            },
+          });
+
+          expect(queryByText('Is authorized: Yes')).toBeInTheDocument();
+        });
+      });
+      describe('when general permissions is view', () => {
+        describe('when data fence permission is manage', () => {
+          it('should indicate as authorized', () => {
+            const { queryByText } = testRender({
+              shouldMatchSomePermissions: true,
+              allAppliedPermissions: [{ name: 'canManageOrders', value: true }],
```

Comment:
but the describe says `when general permissions is view`

Reply:
this is refactored now.